pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -148,10 +148,16 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 			field.Forbidden(specPath.Child("datacenterRef"), fmt.Sprintf("field is immutable %v", new.Spec.DatacenterRef)))
 	}
 
-	if !new.Spec.ClusterNetwork.Equal(&old.Spec.ClusterNetwork) {
+	if !new.Spec.ClusterNetwork.Services.Equal(&old.Spec.ClusterNetwork.Services) {
 		allErrs = append(
 			allErrs,
-			field.Forbidden(specPath.Child("ClusterNetwork"), fmt.Sprintf("field is immutable %v", new.Spec.ClusterNetwork)))
+			field.Forbidden(specPath.Child("ClusterNetwork.Services"), fmt.Sprintf("field is immutable %v", new.Spec.ClusterNetwork)))
+	}
+
+	if !new.Spec.ClusterNetwork.Pods.Equal(&old.Spec.ClusterNetwork.Pods) {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(specPath.Child("ClusterNetwork.Pods"), fmt.Sprintf("field is immutable %v", new.Spec.ClusterNetwork)))
```

Comment:
I need to verify which other fields are immutable.

Reply:
`Nodes` is also immutable per @mitalipaygude. I didn't have a chance to deep dive into `ClusterNetwork.DNS` and the person who wrote that component is no longer on the team, so I was planning to make it immutable until we heard otherwise. There might be more info if you deep dive the code but I haven't had a chance to.

Reply:
I would have to test to confirm the upgrade behavior since I added this feature a while back, but I think we should be safe to say that `ClusterNetwork.DNS` is mutable. If the user doesn't specify it, it will just use the default `resolv.conf` file.

Reply:
Made `Nodes` and `DNS` immutable. I don't see any reason why they should be, but let's start with this and we can remove the protection as needed.